src/inc/msquic.h

```diff
@@ -276,6 +276,54 @@ typedef struct QUIC_NEW_CONNECTION_INFO {
     const char* ServerName;
 } QUIC_NEW_CONNECTION_INFO;
 
+#define QUIC_CIPHER_SUITE_TLS_AES_128_GCM_SHA256          0x1301
+#define QUIC_CIPHER_SUITE_TLS_AES_256_GCM_SHA384          0x1302
+#define QUIC_CIPHER_SUITE_TLS_CHACHA20_POLY1305_SHA256    0x1303
+
+#ifdef CALG_AES_128
+#define QUIC_ALG_AES_128       CALG_AES_128
+#else
+#define QUIC_ALG_AES_128       0x660E
+#endif
```

Comment:
I'm not a super big fan of redefining these all in this way. What about if we just used the values directly and defined them if they don't exist (thus standardizing on the existing Windows value):
```
#ifndef CALG_AES_128
#define CALG_AES_128 0x660E
#endif
```

Reply:
That gets a bit weird, because some of them don't have a matching value in schannel, and I don't want to collide with them when they add chacha20.

Reply:
Once it's already defined by Windows, then your `#ifndef` gets disabled. Also, why isn't there a value for ChaCha20? There has to be one, right? What is Schannel returning in this case (they supposedly support ChaCha20)?

Reply:
I couldn't find a single ChaCha20 definition in schannels headers.

Reply:
If the ifdef gets disabled to a different value, thats a silent ABI breaking change. Definitely NOT ok with that.

Reply:
I agree we shouldn't invent a new value ourselves. We need to find the Windows value.

Reply:
Also, just like we did for address family, if you want to be a good cross platform API, building your own abstraction level really is much better from a usability perspective. The bit of defines here is worth it IMO.

Reply:
Well, if you use the address family as the example, these should go in the platform specific headers.

Reply:
Ok, I'm fine with that.

Reply:
Moving to enums made this be solved in a different way. So we're good there.